include/swift/Sema/ConstraintSystem.h

```diff
@@ -3759,6 +3759,11 @@ class ConstraintSystem {
                                        RememberChoice_t rememberChoice,
                                        ConstraintLocatorBuilder locator,
                                        ConstraintFix *compatFix = nullptr);
+  
+  /// Add a materialize constraint for a pack expansion.
```

Comment:
I'm not sure whether we want to mention the name of the constraint in the method, it would be better to say `pack materialization constraint` or something like to expose the meaning but not the implementation and `returns a type variable that represents a materialized ...`

Reply:
Good point. Inspiration for Sima:

> Add a pack materialization constraint between a fresh type variable and the given type, and return the type variable, which represents the left-hand side. A pack materialization constraint requires that the left-hand side is a tuple with a single unlabeled element matching the right-hand side, and that the right-hand side a pack expansion type.